src/test/kotlin/io/github/nomisrev/routes/ProfileRouteSpec.kt

```diff
@@ -0,0 +1,53 @@
+package io.github.nomisrev.routes
+
+import io.github.nomisrev.env.Dependencies
+import io.github.nomisrev.service.RegisterUser
+import io.github.nomisrev.withServer
+import io.kotest.assertions.arrow.core.shouldBeRight
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.shouldBe
+import io.ktor.client.call.body
+import io.ktor.client.plugins.resources.get
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.contentType
+
+class ProfileRouteSpec :
```

Comment:
We're missing a test case for when the path parameter is missing, I think that would've resulted `MissingRequestParameterException` case.

Reply:
I assume that the endpoint that we'd want to hit/test is `api/profiles/` and assert that our [new validation](https://github.com/nomisRev/ktor-arrow-example/pull/186/files#diff-fd19be7abb78848f1cf6bc8da926adafd98399c75cf49662e9a8dcc29f7f3cb0R57) is triggered

However, I'm having difficulties actually hitting the above endpoint with the current `Resource` setup, I get 404 not found.

I've read through the [docs](https://ktor.io/docs/routing-in-ktor.html#path_parameter), and tried with the `?` approach, but with no luck.

Do you have any suggestions?

Reply:
Strange, I made a small change and it's working fine for me.

Reply:
I'm probably missing something obvious, I've tried various things to trigger the `api/profiles` logic.

The username implementation is marked as optional and a placeholder test that fails on my side is added [here](https://github.com/nomisRev/ktor-arrow-example/pull/186/commits/64dbe4d7141fe837da867733f527b93f08cd4b90)

I've tried also to write it without the `@Resource` implementation so that I just hit `/api/profiles`, but with no luck 🤷